### PR TITLE
Add support for WSL in vscode

### PR DESCRIPTION
**Pull request by s-h-a-d-o-w**
Whenever `ps` is run in a WSL window in vscode, it prints an error (a "false positive" in a way), causing to `pidtree` fail. 

This work around here is also [used in vscode itself](https://github.com/microsoft/vscode/blob/6f26fa19db7e4cd2af57f52fd7009e715c3d7aab/src/vs/base/node/ps.ts#L241).

I don't know whether you want to introduce something like this. As tiny of a change and as simple as it may be, I would understand it if you were to argue that you don't want to start including workarounds for such specific scenarios.
But in case the vscode team [either doesn't want to or is unable to address this at the source](https://github.com/microsoft/vscode/issues/145354) (this problem has existed for years), this would obviously enable `pidtree` to be used for tooling in that environment as well. (I recently contributed an [improvement to the concurrency behavior of lint-staged](https://github.com/okonet/lint-staged/pull/1117) that uses `pidtree` and that I ironically haven't been able to benefit from because I happen to use WSL...)

### CODE REVIEW

**Comment by huan** [comment]
Hi @s-h-a-d-o-w ,

Thanks for adding this to the project! I think it's fine because VSCode code is using it too.

Could you please add a unit test to make sure this works in our code base so that we can have enough confidence to approve it?

Thank you very much.

**Comment by s-h-a-d-o-w** [comment]
Done. Apologies for missing that!

I hope they are alright because... I'm not a fan of these redundancies myself. But looking at the already existing tests and skimming the mockery docs (as well as attempting to remove some of those redundancies), this seemed to me like a conscious decision by the original author and so I ended up following suit.

**Comment by huan** [comment]
Thanks for adding the unit test!

I think this PR is good and I'd like to approve this PR after the CI turns green.

**Review by huan — APPROVED** [review]
LGTM

@simonepri

**Comment by juliomrc** [comment]
Hey! Thank you for this fix.
We are also being affected by this issue. Any prediction on when this will be merged and published?

Thanks a lot!

**Comment by huan** [comment]
@juliomrc could you please give an approval to this PR to help us confirm this code is ok?

@simonepri I think this PR will be good to go, and I will merge it if you do not against it and after it get another approval.

**Review by juliomrc — APPROVED** [review]
LGTM.

**Comment by juliomrc** [comment]
@huan thanks for the quick action.
Will you also release the patch to npm?

**Comment by simonepri** [comment]
Sorry everybody for being so late to the party!

I took a look at the PR, and I'm okay with implementing a workaround for this issue, but I'm not sure I like how we implemented it in this PR as it can cause us to ignore more than just the `bogus screen` error.

I took the liberty to change the code to strip the message from the error stream instead, so that we can ignore only that particular error message.

Please take a look, and confirm this will work as intended:
https://github.com/simonepri/pidtree/commit/4fa79c596d94998ec666639bbe4623432e6a1349

I'll release a new version as soon this is confirmed.
Thanks a lot for the help!

**Comment by simonepri** [comment]
I released a new 0.6.0 version, let me know if this solves the issue.

**Comment by juliomrc** [comment]
This fix works as intended and the edge case integration with lint-staged no longer produces a problem.
Thank you!

**Comment by s-h-a-d-o-w** [comment]
@simonepri 

> it can cause us to ignore more than just the bogus screen error.

That's fair but... on the one hand, I wonder whether `ps` ever produces other errors when this one occurs (but I'm fine with your adjustments, so I will not invest the time looking into this) and on the other, since you are checking for a more specific error message it is obviously more brittle to changes (not that that's likely after all these years...).

That said...

> Thanks a lot for the help!

... and thanks to you for making WSL users happy! I wasn't sure whether it would happen since the last release here was a long time ago (I guess there simply haven't been any issues - which is awesome), so I'm really grateful. 🙂

**Pull request merged**
